core/storage/statistics/gae_models.py

```diff
@@ -423,6 +423,48 @@ def create(cls, exp_id, exp_version, state_name, session_id,
         complete_event_entity.put()
 
 
+class RateExplorationEventLogEntryModel(base_models.BaseModel):
+    """An event triggered by a learner rating the exploration.
+
+    Event schema documentation
+    --------------------------
+    V1:
+        event_type: 'rate_exploration'
+        exploration_id: id of exploration which is being rated
+        rating: value of rating assigned to exploration
+    """
+    # This value should be updated in the event of any event schema change.
+    CURRENT_EVENT_SCHEMA_VERSION = 1
+
+    # Which specific type of event this is
+    event_type = ndb.StringProperty(indexed=True)
+    # Id of exploration currently being played.
+    exploration_id = ndb.StringProperty(indexed=True)
+    # Value of rating assigned
+    rating = ndb.FloatProperty(indexed=True)
+    # The version of the event schema used to describe an event of this type.
+    # Details on the schema are given in the docstring for this class.
+    event_schema_version = ndb.IntegerProperty(
+        indexed=True, default=CURRENT_EVENT_SCHEMA_VERSION)
+
+    @classmethod
+    def get_new_event_entity_id(cls, exp_id):
+        timestamp = datetime.datetime.utcnow()
+        return cls.get_new_id('%s:%s' % (
+            utils.get_time_in_millisecs(timestamp),
+            exp_id))
+
+    @classmethod
+    def create(cls, exp_id, rating):
```

Comment:
I think this has to depend also on user_id, since ratings are associated with an exploration and a user.